src/Avalonia.Controls.DataGrid/Collections/DataGridCollectionView.cs

```diff
@@ -0,0 +1,4315 @@
+﻿// (c) Copyright Microsoft Corporation.
+// This source is subject to the Microsoft Public License (Ms-PL).
+// Please see http://go.microsoft.com/fwlink/?LinkID=131993 for details.
+// All other rights reserved.
+
+using Avalonia.Controls.Utils;
+using Avalonia.Utilities;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace Avalonia.Collections
+{
+    /// <summary>
+    /// Event argument used for page index change notifications. The requested page move
+    /// can be canceled by setting e.Cancel to True.
+    /// </summary>
+    public sealed class PageChangingEventArgs : CancelEventArgs
+    {
+        /// <summary>
+        /// Constructor that takes the target page index
+        /// </summary>
+        /// <param name="newPageIndex">Index of the requested page</param>
+        public PageChangingEventArgs(int newPageIndex)
+        {
+            NewPageIndex = newPageIndex;
+        }
+
+        /// <summary>
+        /// Gets the index of the requested page
+        /// </summary>
+        public int NewPageIndex
+        {
+            get;
+            private set;
+        }
+    }
+
+    /// <summary>Defines a method that enables a collection to provide a custom view for specialized sorting, filtering, grouping, and currency.</summary>
+    internal interface ICollectionViewFactory
+    {
+        /// <summary>Returns a custom view for specialized sorting, filtering, grouping, and currency.</summary>
+        /// <returns>A custom view for specialized sorting, filtering, grouping, and currency.</returns>
+        IDataGridCollectionView CreateView();
+    }
+
+    /// <summary>
+    /// PagedCollectionView view over an IEnumerable.
+    /// </summary>
+    public sealed class CollectionViewBase : IDataGridCollectionView, IDataGridEditableCollectionView, INotifyPropertyChanged //IPagedCollectionView, 
```

Comment:
Why is this named `CollectionViewBase` when it's sealed? `Base` suggests it should be the base class for something.

Also there's commented out code with `IPagedCollectionView`.